src/core/system.js

```diff
@@ -69,7 +65,7 @@ System.prototype = {
    */
   updateProperties: function (rawData) {
     var oldData = this.data;
-    if (!Object.keys(schema).length) { return; }
+    if (!Object.keys(this.schema).length) { return; }
```

Comment:
Huh, you found an issue it seems, so before that was always false?
And now it's returning true for single property system if I understand it? That's okay?

Reply:
Before it was actually always true, since `schema` was imported and in scope, see [system.js#L2](https://github.com/aframevr/aframe/blob/master/src/core/system.js#L2).

Now it's only true if the schema has no keys, meaning there should be nothing to do anyway. I believe this was the original intention behind the guard. For any system that is single-property or multi-property nothing changes, only those that don't have any properties, and thus don't expect any data, will now skip the data parsing.

Reply:
Right, I completely misread the condition. So okay all good. I find it hard to process in my mind what `!Object.keys(this.schema).length` means. `Object.keys(this.schema).length === 0` would be clearer to me.